// frontend/src/pages/ChatAnalyzePage/lib/mockApi.ts
import type { AnalysisResult } from '../types/analyze.types'
const getRandomDelay = () =>
  Math.floor(Math.random() * (1500 - 1200 + 1) + 1200)
const mockPlantData: AnalysisResult[] = [
  {
    plant: {
      commonName: 'Cà chua',
      scientificName: 'Solanum lycopersicum',
    },
    disease: {
      name: 'Bệnh đốm lá sớm',
      description:
        'Bệnh đốm lá sớm thường gây ra các đốm tròn màu nâu trên lá với các vòng đồng tâm. Bệnh phát triển từ lá già và lan lên trên, có thể gây rụng lá và giảm năng suất.',
    },
    confidence: 0.85,
    care: [
      'Loại bỏ lá bị bệnh và tiêu hủy, không để trên đất.',
      'Tưới nước vào gốc, tránh làm ướt lá.',
      'Sử dụng thuốc trừ nấm có chứa chlorothalonil hoặc mancozeb.',
      'Luân canh cây trồng, tránh trồng cà chua liên tiếp trên cùng một mảnh đất.',
    ],
    products: [
      {
        name: 'Thuốc trừ nấm Daconil',
        imageUrl:
          'https://images.unsplash.com/photo-1471193945509-9ad0617afabf?ixlib=rb-1.2.1&auto=format&fit=crop&w=400&h=400&q=80',
        price: '180.000₫',
        note: 'Phòng trừ bệnh đốm lá và nhiều loại nấm khác',
      },
      {
        name: 'Phân bón hữu cơ vi sinh',
        imageUrl:
          'https://images.unsplash.com/photo-1585314062604-1a357de8b000?ixlib=rb-1.2.1&auto=format&fit=crop&w=400&h=400&q=80',
        price: '75.000₫',
        note: 'Tăng cường sức đề kháng cho cây trồng',
      },
      {
        name: 'Bình phun thuốc cầm tay',
        imageUrl:
          'https://images.unsplash.com/photo-1590682680695-43b964a3ae17?ixlib=rb-1.2.1&auto=format&fit=crop&w=400&h=400&q=80',
        price: '250.000₫',
        note: 'Phun thuốc đều và tiết kiệm',
      },
    ],
    imageInsights: {
      previewUrl:
        'https://images.unsplash.com/photo-1592496001020-d31bd830651f?ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80',
      boxes: [
        { x: 20, y: 30, width: 25, height: 25, label: 'Đốm lá' },
        { x: 60, y: 45, width: 20, height: 20, label: 'Đốm lá' },
      ],
    },
  },
  {
    plant: {
      commonName: 'Xoài',
      scientificName: 'Mangifera indica',
    },
    disease: {
      name: 'Bệnh thán thư',
      description:
        'Bệnh thán thư trên xoài gây ra các đốm đen trên lá, hoa và quả. Trên quả, bệnh tạo ra các vết thối đen, làm giảm chất lượng và năng suất.',
    },
    confidence: 0.78,
    care: [
      'Cắt tỉa và tiêu hủy các cành lá bị bệnh.',
      'Phun thuốc phòng trừ nấm trước mùa mưa.',
      'Tránh làm tổn thương quả khi thu hoạch và vận chuyển.',
      'Bảo quản quả ở nơi khô ráo, thoáng mát.',
    ],
    products: [
      {
        name: 'Thuốc trừ nấm Carbendazim',
        imageUrl:
          'https://images.unsplash.com/photo-1591753168641-1c93c2adcef3?ixlib=rb-1.2.1&auto=format&fit=crop&w=400&h=400&q=80',
        price: '120.000₫',
        note: 'Phòng trừ hiệu quả bệnh thán thư',
      },
      {
        name: 'Kéo cắt tỉa chuyên dụng',
        imageUrl:
          'https://images.unsplash.com/photo-1614334636858-92a6e3f44cfd?ixlib=rb-1.2.1&auto=format&fit=crop&w=400&h=400&q=80',
        price: '220.000₫',
        note: 'Cắt tỉa dễ dàng, sạch vết cắt',
      },
    ],
    imageInsights: {
      previewUrl:
        'https://images.unsplash.com/photo-1601493700631-2b16ec4b4716?ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80',
      boxes: [{ x: 35, y: 40, width: 30, height: 20, label: 'Thán thư' }],
    },
  },
  {
    plant: {
      commonName: 'Lúa',
      scientificName: 'Oryza sativa',
    },
    disease: null,
    confidence: 0.92,
    care: [
      'Duy trì mực nước phù hợp theo giai đoạn sinh trưởng.',
      'Bón phân cân đối N-P-K theo khuyến cáo.',
      'Phòng trừ sâu bệnh theo nguyên tắc IPM.',
      'Thu hoạch đúng thời điểm khi 85-90% hạt chín.',
    ],
    products: [
      {
        name: 'Phân bón NPK chuyên lúa',
        imageUrl:
          'https://images.unsplash.com/photo-1632401495112-0d0c36eb1a0a?ixlib=rb-1.2.1&auto=format&fit=crop&w=400&h=400&q=80',
        price: '280.000₫/bao',
        note: 'Cân đối dinh dưỡng cho lúa phát triển tốt',
      },
      {
        name: 'Thuốc trừ cỏ lúa an toàn',
        imageUrl:
          'https://images.unsplash.com/photo-1620552410974-e838f3c41b8d?ixlib=rb-1.2.1&auto=format&fit=crop&w=400&h=400&q=80',
        price: '150.000₫',
        note: 'Diệt cỏ hiệu quả, an toàn cho lúa',
      },
    ],
    imageInsights: {
      previewUrl:
        'https://images.unsplash.com/photo-1536431311719-398b6704d4cc?ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80',
      boxes: [],
    },
  },
]
export const mockApi = {
  analyze: async (): Promise<AnalysisResult> => {
    // Simulate API delay
    await new Promise((resolve) => setTimeout(resolve, getRandomDelay()))
    // Return a random mock result
    const randomIndex = Math.floor(Math.random() * mockPlantData.length)
    return mockPlantData[randomIndex]
  },
}
