// frontend/src/pages/ChatAnalyzePage/components/analysis/ImageAnalysisCard.tsx
import React from 'react'
import { ImageIcon } from 'lucide-react'
import type { ImageInsights } from '../../types/analyze.types'
interface ImageAnalysisCardProps {
  imageInsights: ImageInsights | undefined
}
export const ImageAnalysisCard: React.FC<ImageAnalysisCardProps> = ({
  imageInsights,
}) => {
  if (!imageInsights) {
    return (
      <div className="bg-white rounded-2xl shadow-sm p-6">
        <h2 className="text-lg font-medium mb-4">Phân tích hình ảnh</h2>
        <div className="flex flex-col items-center justify-center p-8 border-2 border-dashed border-gray-300 rounded-xl bg-gray-50">
          <ImageIcon size={32} className="text-gray-400 mb-2" />
          <p className="text-gray-500">Chưa có hình ảnh để phân tích</p>
        </div>
      </div>
    )
  }
  return (
    <div className="bg-white rounded-2xl shadow-sm p-6">
      <h2 className="text-lg font-medium mb-4">Phân tích hình ảnh</h2>
      <div className="relative">
        <img
          src={imageInsights.previewUrl}
          alt="Plant analysis"
          className="w-full rounded-xl"
        />
        {imageInsights.boxes &&
          imageInsights.boxes.map((box, index) => (
            <div
              key={index}
              className="absolute border-2 border-red-500 bg-red-500/20"
              style={{
                left: `${box.x}%`,
                top: `${box.y}%`,
                width: `${box.width}%`,
                height: `${box.height}%`,
              }}
            >
              <div className="absolute -top-6 left-0 bg-red-500 text-white text-xs py-0.5 px-2 rounded">
                {box.label}
              </div>
            </div>
          ))}
      </div>
      <div className="mt-4">
        <p className="text-sm text-gray-700">
          Các vùng được đánh dấu cho thấy các dấu hiệu bất thường trên cây trồng
          cần chú ý.
        </p>
      </div>
    </div>
  )
}
